azure-kusto-data/tests/test_kusto_client.py

```diff
@@ -341,43 +327,186 @@ def test_admin_then_query(self, mock_post):
         assert response[2].table_kind == WellKnownDataSet.QueryCompletionInformation
         assert response[3].table_kind == WellKnownDataSet.TableOfContents
 
-    @patch("requests.Session.post", side_effect=mocked_requests_post)
-    def test_dynamic(self, mock_post):
-        """Tests dynamic responses."""
-        client = KustoClient("https://somecluster.kusto.windows.net")
-        query = """print dynamic(123), dynamic("123"), dynamic("test bad json"),"""
-        """ dynamic(null), dynamic('{"rowId":2,"arr":[0,2]}'), dynamic({"rowId":2,"arr":[0,2]})"""
-        row = client.execute_query("PythonTest", query).primary_results[0].rows[0]
+    @staticmethod
+    def _assert_dynamic_response(row):
         assert isinstance(row[0], int)
         assert row[0] == 123
-
         assert isinstance(row[1], str)
         assert row[1] == "123"
-
         assert isinstance(row[2], str)
         assert row[2] == "test bad json"
-
         assert row[3] is None
-
         assert isinstance(row[4], str)
         assert row[4] == '{"rowId":2,"arr":[0,2]}'
-
         assert isinstance(row[5], dict)
         assert row[5] == {"rowId": 2, "arr": [0, 2]}
 
+
+class KustoClientTestsSync(unittest.TestCase, KustoClientMixin):
```

Comment:
Please revert this. I'm not a fan of splitting assertion code from tests.
There are other ways of reducing code.
(also, this is unrelated to the PR)

Reply:
It is related because async flow uses the same asserts